psi4/src/psi4/dfocc/dfocc.cc

```diff
@@ -109,40 +109,8 @@ void DFOCC::common_init() {
     // title
     title();
 
-    //   Tying orbital convergence to the desired e_conv,
-    //   particularly important for the same numerical frequencies by energy
-    //   These have been determined by linear fits to a step fn
-    //   based on e_conv on limited numerical tests.
-    //   The printed value from options_.print() will not be accurate
-    //   since newly set orbital conv is not written back to options
-    if (orb_opt_ == "TRUE") {
-        if (options_["RMS_MOGRAD_CONVERGENCE"].has_changed()) {
-            tol_grad = options_.get_double("RMS_MOGRAD_CONVERGENCE");
-        } else {
-            double temp;
-            temp = (-0.9 * std::log10(tol_Eod)) - 1.6;
-            if (temp < 4.0) {
-                temp = 4.0;
-            }
-            tol_grad = pow(10.0, -temp);
-            // tol_grad = 100.0*tol_Eod;
-            outfile->Printf("\tRMS orbital gradient is changed to : %12.2e\n", tol_grad);
-        }
-
-        // Determine the MAXIMUM MOGRAD CONVERGENCE
-        if (options_["MAX_MOGRAD_CONVERGENCE"].has_changed()) {
-            mograd_max = options_.get_double("MAX_MOGRAD_CONVERGENCE");
-        } else {
-            double temp2;
-            temp2 = (-0.8 * std::log10(tol_grad)) - 0.5;
-            if (temp2 < 3.0) {
-                temp2 = 3.0;
-            }
-            mograd_max = pow(10.0, -temp2);
-            // mograd_max = 10.0*tol_grad;
-            outfile->Printf("\tMAX orbital gradient is changed to : %12.2e\n", mograd_max);
-        }
-    }  // end if (orb_opt_ == "TRUE")
```

Comment:
@bozkaya Can you review this in light of the issue?

Reply:
Was this resetting of the mograd_conv keywords causing issues, @JonathonMisiewicz? It's admittedly strange, but I don't think it's harmful. The read_options comment should probably note that the defaults are tied to e_convergence.

Reply:
It was causing an issue, but this was probably an issue of expected behavior. I assumed that since [we bothered to set default convergence criteria specifically for the dfocc module](https://github.com/psi4/psi4/blob/master/psi4/src/read_options.cc#L2758-L2762) that these were supposed to be honored, and that when dfocc said it was _changing_ the convergence criteria, it meant that it was overriding the defaults which were actually supposed to be the defaults.

It sounds like the default behavior is instead supposed to be handled internally by `dfocc` and `occ`, not by the defaults in `read_options.cc`. Is that correct? If so, I can restore the code and make this much clearer. The options need to be better documented, the code in occ and dfocc needs to be documented better, and we should just set the `read_options` defaults that are actually placeholders to values that are obviously placeholders, and not things that look like actual defaults.

By the way, the `dfomp-1` and `dfomp2-2` tests don't actually converge energy to e-6 with the defaults `dfocc` sets.

Reply:
dfocc is doing the right thing by the user by obeying the [user's specific directions](https://github.com/psi4/psi4/pull/1772/files#diff-3f4d38c9abf7b05dc3f5c0296fb1dac5L133-L134) wrt mograd. liboptions is limited (not to its discredit) in that it can only set up independent defaults for options, whereas sometimes [modules need tied options](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/detci/params.cc#L145-L184). This particular tying is peculiar, but so long as it's respecting the user's override, it's ok imo.

What you mention about tests not converging to 1.e-6 is bad. You fixed it through the deletion https://github.com/psi4/psi4/pull/1772/files#diff-550445b03966d0041b9fa2c2fb58e4c8L278-L281, right?

Reply:
Okay, I'll restore the code in question but make it clearer that is what it's actually doing.

No, [deleting those lines](https://github.com/psi4/psi4/pull/1772/files#diff-550445b03966d0041b9fa2c2fb58e4c8L278-L281) didn't fix it. The problem is that the orbital steps converge so slowly in `dfocc` that even though subsequent steps differ by less than 1.e-6, we're still 2.e-6 or so away from the converged result. I just amped up the convergence criteria so the tests passed.

Any further comments on the test convergence before I commit review corrections?

Reply:
Ah, I think I see what you mean. I tried out the deletion and it "worked", but I was only looking at `DE` at face value. In SCF, `D_CONVERGENCE` is harder to achieve than `DE`, so the extra iterations to get the former, effectively make sure the latter is not just nominally satisfied but is w/i that delta of the converged energy. Whereas the tied gradient and energy convcrit that dfocc sets up is so loose on the former that the latter is the harder to converge even to nominal satisfaction and doesn't get w/i the DE of the actual answer. Does this sound like your assessment? Can we just add 1 to the grad power that the linear regression spits out? I think actually converging is more important than deceptive minimization of niter here.

Reply:
Yeah, for the default convergence settings, it's harder to get the nominal energy convergence than the `max_mograd_convergence`, so we may or may not get within the `DE` of the actual energy.

Just increasing the gradient power should work fine for now. I have future plans to improve `dfocc` orbital convergence (my Psi4Numpy script doesn't have the orbital convergence problems `dfocc` has).

Reply:
Sounds good. My vote's for increasing the gradient power.

Reply:
_siiiiiigh_ And it turns out that `occ` and `dfocc` have completely different maps to go from energy convergence to orbital convergence. Any objections to me making `occ` use `dfocc` settings as well? DF should not change convergence thresholds.

Reply:
No objections -- hack away.